src/ConnectionPool.jl

```diff
@@ -264,10 +270,32 @@ function IOExtras.closeread(t::Transaction)
     notify(t.c.readdone)                          ;@debug 2 "✉️  Read done:  $t"
     notify(poolcondition)
 
+    if !isbusy(t.c)
+        @schedule monitor_idle_connection(t.c)
```

Comment:
maybe worth doing `let c = t.c; @schedule monitor_idle_connection(t.c); end`, so we don't capture all of `t`. But afaik, there's no `finalizers` on `t`, or other way to see this difference.

Reply:
Yeah, `t` is an immutable struct containing just `c` and an `Int`, so probably not much difference.
But it's a good reminder to consider what is being captured in closures. I have a feeling I should be more cognisant of that more often...